### PR TITLE
replace use of bitcoin::utils::misc::hex_bytes with hex::decode

### PR DESCRIPTION
**Pull request by savil**
as requested in #84 discussion
https://github.com/rust-bitcoin/rust-lightning/pull/84#discussion_r205508845

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Thanks! Also in fuzz/fuzz_targets. If you're gonna change every line anyway why not just write out "hex::decode"? Otherwise just import it as hex_bytes so you dont have to change every line.

**Comment by savil** [comment]
ah, yeah, forgot about fuzz directory. will do, hold on.

hmm - I guess, I was trying to match the existing code style of importing on top. I can change to `::hex::decode` no problem.

Slight preference to avoid `hex_bytes` because then when reading code some files will say `hex_decode`/`::hex::decode` and some `hex_bytes`, and newbies will have to lookup the import line to realize they are the same things. Not a big deal, but a minor cognitive dissonance.

**Comment by savil** [comment]
in `fuzz/`, what did you have in mind? `git grep "hex_bytes"` has no results.

**Comment by TheBlueMatt** [comment]
In fuzz there's a bunch of extend_vec_from_hex's that could get replaced. import hex; and hex::decode seems good to me, or ::hex::decode, though I find that less readable.

**Comment by savil** [comment]
ah, got it (on both points).

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -16,12 +16,13 @@ afl_fuzz = ["afl"]
 honggfuzz_fuzz = ["honggfuzz"]
 
 [dependencies]
+afl = { version = "0.3", optional = true }
 lightning = { path = "..", features = ["fuzztarget"] }
 bitcoin = { version = "0.13", features = ["fuzztarget"] }
-secp256k1 = { version = "0.9", features=["fuzztarget"] }
-rust-crypto = "0.2"
+hex = "0.3.2"
```
Comment (TheBlueMatt) [inline fuzz/Cargo.toml]: I think usually its best to just specify 0.3, as minor bumps we'd want to take as bugfixes, no?

Comment (savil) [reply fuzz/Cargo.toml]: fair point!

**Pull request merged**